test/sharness/t0054-dag-car-import-export.sh

```diff
@@ -0,0 +1,25 @@
+#!/usr/bin/env bash
+#
+
+test_description="Test car file import/export functionality"
+
+. lib/test-lib.sh
+
+test_init_ipfs
+
+
+echo "Error: merkledag: not found (currently offline, perhaps retry after attaching to the network)" > offline_fetch_error_expected
+
+test_expect_success "basic offline export of nonexistent cid" '
+  ! ipfs dag export QmYwAPJXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX 2> offline_fetch_error_actual
+'
+
+test_expect_success "correct error" '
+  test_cmp offline_fetch_error_expected offline_fetch_error_actual
+'
+
+test_expect_success "basic offline export of 'getting started' dag" '
+  ipfs dag export QmS4ustL54uo8FzR9455qaxZwuMiUhyvMcX9Ba8nUH4uVv >/dev/null
+'
+
+test_done
```

Comment:
The code looks good. But this will need some tests.

Reply:
There are are way more tests in the import part of the PR, because they mesh together ( import shuffled / export / check that the original converge )

Plus there are tests in the original that *didn't* get split correctly.

TLDR: let's merge this + the progress part **if the code is good**, then I will re-bae the importer and recheck the tests / fix conflicts etc.

Reply:
Well, I want to see passing tests somewhere first (the tests on the original PR look like they're currently failing). But we can base the import PR on this one and add the export tests there.

Reply:
@Stebalien ok, making it so. There are also *more* tests in the progress bar PR, perhaps check this quick ( it's small ) and I'll work on rebasing import on all that + master

So you can look imto import afresh as a standalone diff

Reply:
Also when you say tests: sharness tests are enough right? Or you have something more in mind?

Reply:
Just sharness.